pkg/controllers/routing/network_routes_controller.go

```diff
@@ -341,6 +342,49 @@ func (nrc *NetworkRoutingController) updateCNIConfig() {
 			glog.Fatalf("Failed to insert `subnet`(pod CIDR) into CNI conf file: %s", err.Error())
 		}
 	}
+
+	err = nrc.autoConfigureMTU()
```

Comment:
Configuring the MTU in the CNI only changes the MTU for the interface that is placed inside the network namespace on the container side. You still need to adjust the kube-bridge interface as well, otherwise the packet will get adjusted when exiting or entering the bridge interface on the host side.

For this, I made the following adjustment to network_routes_controller.Run():
```
	// create 'kube-bridge' interface to which pods will be connected
	kubeBridgeIface, err := netlink.LinkByName("kube-bridge")
	if err != nil && err.Error() == IfaceNotFound {
		linkAttrs := netlink.NewLinkAttrs()
		linkAttrs.Name = "kube-bridge"
		bridge := &netlink.Bridge{LinkAttrs: linkAttrs}
		if err = netlink.LinkAdd(bridge); err != nil {
			glog.Errorf("Failed to create `kube-router` bridge due to %s. Will be created by CNI bridge plugin when pod is launched.", err.Error())
		}
		kubeBridgeIf, err := netlink.LinkByName("kube-bridge")
		if err != nil {
			glog.Errorf("Failed to find created `kube-router` bridge due to %s. Will be created by CNI bridge plugin when pod is launched.", err.Error())
		}
		err = netlink.LinkSetUp(kubeBridgeIf)
		if err != nil {
			glog.Errorf("Failed to bring `kube-router` bridge up due to %s. Will be created by CNI bridge plugin at later point when pod is launched.", err.Error())
		}
		kubeBridgeIface, err = netlink.LinkByName("kube-bridge")
		if err != nil {
			glog.Error("Even after attempting to bring kube-router bridge up, we still weren't able to fetch interface: %s", err.Error())
		}
	}
	mtu, err := getMTUFromNodeIP(nrc.nodeIP, nrc.enableOverlays)
	if err != nil {
		glog.Errorf("Failed to find MTU for node IP: %s for intelligently setting the kube-bridge MTU due to %s.", nrc.nodeIP, err.Error())
	}
	if mtu > 0 {
		glog.Infof("Setting MTU of kube-bridge interface to: %d", mtu)
		err = netlink.LinkSetMTU(kubeBridgeIface, mtu)
		if err != nil {
			glog.Errorf("Failed to set MTU for kube-bridge interface due to: %s", err.Error())
		}
	} else {
		glog.Infof("Not setting MTU of kube-bridge interface")
	}
```

Without this adjustment, when running iperf, I saw no actual improvements in performance of the interface.